File: 1584. Min Cost to Connect All Points/1584. Min Cost to Connect All Points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minCostConnectPoints = function (points) {
  const n = points.length;
  let cost = 0;
  const seen = new Set();
  const minHeap = new MinPriorityQueue({ compare: (a, b) => a[0] - b[0] });
  minHeap.enqueue([0, 0]);

  while (seen.size < n) {
    const [dist, i] = minHeap.dequeue();

    if (seen.has(i)) {
      continue;
    }

    seen.add(i);
    cost += dist;

    const [x, y] = points[i];

    for (let j = 0; j < n; j++) {
      if (!seen.has(j)) {
        const [otherX, otherY] = points[j];
        const manDist = Math.abs(x - otherX) + Math.abs(y - otherY);
        minHeap.enqueue([manDist, j]);
      }
    }
  }

  return cost;
};

File: 1584. Min Cost to Connect All Points.py
class Solution:
    def minCostConnectPoints(self, points: List[List[int]]) -> int:
        cost = 0
        seen = set()
        min_heap = [(0, 0)]

        while len(seen) != len(points):
            c, i = heapq.heappop(min_heap)
            if i in seen:
                continue
            seen.add(i)
            cost += c
            xi, yi = points[i]


            for j in range(len(points)):
                if j not in seen:
                    xj, yj = points[j]
                    dist = abs(xi - xj) + abs(yi - yj)
                    heapq.heappush(min_heap, (dist, j))
        
        return cost
